common/services/user.js

```diff
@@ -14,19 +17,27 @@ function getFullname(token) {
 }
 
 function getLocations(token) {
```

Comment:
It feels like this method has got a few things going on.

Rather than having the line `if (authorities.includes('ROLE_PECS_SUPPLIER'))` inside of this method and it doing two different things what do you think to having it at a higher level?

so for instance in the `app/auth/middleware.js` file we make a decision on what type of auth user this is and send in different items to the `Promise.all`.

Then either in the `common/services/user.js` file or in a new `common/services/user-supplier.js` file we use the common methods:
• `getAuthGroups`
• `getNomisCaseloads`

But have separate methods:
## `getLocations`
```
function getLocations(token) {
  const { auth_source: authSource } = decodeAccessToken(token)

  switch (authSource) {
    case 'auth':
      return getAuthGroups(token).then(getLocationsByNomisAgencyId)
    case 'nomis':
      return getNomisCaseloads(token).then(getLocationsByNomisAgencyId)
    default:
      return Promise.resolve([])
  }
}
```
## `getSupplierLocations`
```
function getSupplierLocations(token) {
  const { authorities = [], auth_source: authSource } = decodeAccessToken(token)

  switch (authSource) {
    case 'auth':
      return getAuthGroups(token).then(([firstGroup]) => getLocationsBySupplierId(firstGroup))
    case 'nomis':
      return getNomisCaseloads(token).then(getLocationsByNomisAgencyId)
    default:
      return Promise.resolve([])
  }
}
```

This then means that these methods are doing just one thing rather than two. Easier to understand at a higher level and easier to test, Theres is a little repetition but I feel that helps with explanation rather than needing to be dry'd up.

What do you think?

Reply:
Yeah I wasn't too sure on it to be honest and didn't like that we're introducing role logic into the service. I like the idea of having separate methods for getting supplier/other user locations.

So maybe we keep that logic in the auth middleware to determine which service method to call.